### PR TITLE
feat: add timer

### PR DESCRIPTION
**Pull request by OlgaSavina**
## Describe your changes

- I have implemented timer to verify email and restore password pages

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?selectedIssue=AG-151)
photo_2023-12-14_16-51-26

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/app/[locale]/verify-email/components/verify-view/verify-view.tsx [hunk 1]
```diff
@@ -30,9 +30,14 @@ export function VerifyView({ email }: Props) {
   const [sendVerificationCode] = useSendVerificationCodeMutation();
   const [verifyEmail] = useVerifyEmailMutation();
 
+  const [remainingTime, setRemainingTime] = useState<number>(180);
+  const [isExpired, setIsExpired] = useState<boolean>(false);
+
   const handleSendCode = useCallback(async (): Promise<void> => {
     try {
       await sendVerificationCode({ email });
+      setRemainingTime(180);
```
Comment (rudenkoserhii) [inline src/app/[locale]/verify-email/components/verify-view/verify-view.tsx]: Magic Numbers

File: src/app/[locale]/verify-email/components/verify-view/verify-view.tsx [hunk 2]
```diff
@@ -66,6 +71,20 @@ export function VerifyView({ email }: Props) {
       setValue('email', email);
     }
   });
+  useEffect(() => {
+    const timerId: NodeJS.Timeout = setInterval(() => {
+      setRemainingTime((prevTime) => (prevTime > 0 ? prevTime - 1 : 0));
+    }, 1000);
```
Comment (rudenkoserhii) [inline src/app/[locale]/verify-email/components/verify-view/verify-view.tsx:78]: Maybe too

File: src/sections/verify-code-view/form.tsx [hunk 3]
```diff
@@ -79,7 +86,9 @@ export default function VerifyCodeForm(): JSX.Element {
     try {
       await sendCode({ email }).unwrap();
 
-      enqueueSnackbar(t('successMessage'), { variant: 'success' });
+      enqueueSnackbar(t('CheckEmail'), { variant: 'success' });
+      setRemainingTime(180);
```
Comment (rudenkoserhii) [inline src/sections/verify-code-view/form.tsx]: And here

File: src/app/[locale]/verify-email/components/verify-view/verify-view.tsx (hunk 1 above)
Comment (OlgaSavina) [reply src/app/[locale]/verify-email/components/verify-view/verify-view.tsx]: fixed

File: src/sections/verify-code-view/form.tsx (hunk 3 above)
Comment (OlgaSavina) [reply src/sections/verify-code-view/form.tsx]: fixed

**Review by rudenkoserhii — APPROVED** [review]

**Review by canterbery — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**